flake.nix

```diff
@@ -0,0 +1,70 @@
+/*
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+{
+  description = "XiangShan";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }@inputs:
+    let
+      overlay = import ./overlay.nix;
+    in
+    flake-utils.lib.eachDefaultSystem
+      (system:
+      let
+        pkgs = import nixpkgs { inherit system; overlays = [ overlay ]; };
+        deps = with pkgs; [
+          git
+          glibcLocales
+          gnumake autoconf automake
+          mill
+          dtc
+          verilator-5016 cmake ninja
+          python3 python3Packages.pip
+          pkgsCross.riscv64-embedded.buildPackages.gcc
+          pkgsCross.riscv64-embedded.buildPackages.gdb
+          openocd
+          circt
+          mold
+          scons
```

Comment:
Is scons required?

Reply:
scons is not required, this is used is gem5.
Will remove.